src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -359,4 +360,95 @@ public void registerUserForCourse(User user, Course course) {
         auditEventRepository.add(auditEvent);
         log.info("User " + user.getLogin() + " has successfully registered for course " + course.getTitle());
     }
+
+    /**
+     * Get the active students for this particular course
+     *
+     * @param courseId the id of the course
+     * @param periodIndex an index indicating which time period, 0 is current week, -1 is one week in the past, -2 is two weeks in the past ...
+     */
+    public Integer[] getCourseStatistics(Long courseId, Integer periodIndex) {
```

Comment:
A suggestion is to change `periodIndex` to an enum so that it's easier to understand/use the method. So something like `TimePeriod.PastWeek`, `TimePeriod.CurrentWeek`,etc.

Reply:
Hm, the issue would be that you have to make a enum for every week. So like pastWeek, 2WeeksAgo, 3WeeksAgo, etc. Since you can click the left/right arrow potentially infinite times, i dont think that an enum is suitable here.

Reply:
Ah ok!